tests/test_binaries.py

```diff
@@ -242,6 +243,21 @@ def test_brk_regression(self):
         with open(os.path.join(workspace, "test_00000000.messages")) as f:
             self.assertIn("finished with exit status: 0", f.read())
 
+    def test_unaligned_mappings(self):
+        # This test ensures that mapping file contents at non page-aligned offsets is possible.
+        filename = os.path.join(os.path.dirname(__file__), 'binaries', 'basic_linux_amd64')
+        with open(filename, 'rb') as f:
+            for addr, size in [
+                (0x1, 0x1000),
+                (0x1, 0x1fffe),
+                (0x1, 0x1ffff),
+                (0xfff, 0x1),
+                (0xfff, 0x2),
+                (0xfff, 0x1000)
+            ]:
+                ret = mmap(f.fileno(), addr, size)
```

Comment:
We should probably assert that `ret` is different than -1 (or to be more specific: `0xffffffffffffffff` or `ctypes.c_uint64(-1).value`.

Reply:
`ret` would be a `ctypes.POINTER(ctypes.c_char)` here, so it would need to be casted back to a `ctypes.c_void_p` to be compared. I have added an assert into the code of mmap instead.

```python
In [1]: import ctypes

In [2]: ctypes.c_void_p(-1) == ctypes.c_void_p(-1)
Out[2]: False
```